django_extensions/db/fields/__init__.py

```diff
@@ -99,6 +99,36 @@ class AutoSlugField(UniqueFieldMixin, SlugField):
     overwrite
         If set to True, overwrites the slug on every save (default: False)
 
+    slugify_function
+        Defines the function which will be used to "slugify" a content
+        (default: :py:func:`~django.template.defaultfilters.slugify` )
+
+    It is possible to provide custom "slugify" function with
+    the ``slugify_function`` attribute in a model class.
+
+    ``slugify_function`` attribute in a model class takes priority over
+    ``slugify_function`` given as an attribute to :py:class:`~AutoSlugField`.
+
+    Example
+
+    .. code-block:: python
+
+        # models.py
+
+        from django.db import models
+
+        from django_extensions.db.fields import AutoSlugField
+
+
+        def my_slugify_function(self, content):
+            return content.replace('_', '-').lower()
+
+        class MyModel(models.Model):
+            slugify_function = my_slugify_function
```

Comment:
What about using `class Meta` to specify the slugify function ?

```python
class MyModel(models.Model):
    title = models.CharField(max_length=42)
    slug = AutoSlugField(populate_from='title')

    class Meta:
        slugify_function = my_slugify_function
```

That feels a bit cleaner to me...

Reply:
Agree

Reply:
I did some research on **Meta** options.
Django has predefined [DEFAULT_NAMES](https://github.com/django/django/blob/master/django/db/models/options.py#L28) which are supposed to be the only attributes you can define within a **Meta** class.

<details><summary>The error if you try to define any custom attribute in <b>Meta</b></summary>
<code>
TypeError: 'class Meta' got invalid attribute(s): autoslugfield_slugify_function
</code>
</details>

The only way to bypass the problem is to redefine those **DEFAULT_NAMES** in a model's module where you are going to use the feature (which seems a bit cumbersome):
```py
from django.db import models
# ...
models.options.DEFAULT_NAMES = models.options.DEFAULT_NAMES + ('autoslugfield_slugify_function',)
```

If we decided to work with Meta, the overall code would look like this:
```py
"""
some models.py module
"""
from django.db import models

from django_extensions.db.models import TitleSlugDescriptionModel
from pytils.translit import slugify


# We have to write this line of code in each 
# module where we decided to use custom "slugify" feature
models.options.DEFAULT_NAMES = models.options.DEFAULT_NAMES + ('autoslugfield_slugify_function',)


class Author(TitleSlugDescriptionModel, models.Model):
    class Meta:
        autoslugfield_slugify_function = slugify
```

Reply:
That definitely will not make it cleaner if we have to patch `models.options.DEFAULT_NAMES`.
Ah well.. it was a nice try :-) 


Think this would work as well right ?
```python
class MyModel(models.Model):
    title = models.CharField(max_length=42)
    slug = AutoSlugField(populate_from='title')

    def slugify_function(self, content):
        return content.replace('_', '-').lower()
```
or
```python
class MyModel(models.Model):
    title = models.CharField(max_length=42)
    slug = AutoSlugField(populate_from='title')

    def slugify_function(self, content):
        return slugify(content)
```

Which would probably be my preferred way of writing it now that `Meta` does not work :-)
Also makes it nice for creating mixin classes etc.

Reply:
> Think this would work as well right ?

Correct.
I altered docs and examples.
Indeed, function looks way more attractive :)